cypress/support/step_definitions/organisation/deleteMember/deleteMember.ts

```diff
@@ -0,0 +1,21 @@
+import { Given, Then, When } from "cypress-cucumber-preprocessor/steps";
+import { getOrganisationId } from "../../../utils";
+
+Given("the user is on the organisation detail page", () => {
+  cy.visit(`/orgs/${getOrganisationId()}/`);
+  cy.findByText("Members").should("be.visible");
+  cy.waitUntil(() => cy.contains("li", "openmrs", { timeout: 10000 }));
+});
+When("the user clicks on the delete member button", () =>{
```

Comment:
Hi @ibacher the issue I was explaining is over here, it clicks the new member but ends up deleting the original user(the old one). Please help me to fix this.

Reply:
So this is actually a bug in the product (yay!). We use [this component](https://github.com/openmrs/openmrs-ocl-client/blob/master/src/apps/organisations/components/OrgMembers.tsx) to render the organisation members including the delete icon. Hence, [the block starting here](https://github.com/openmrs/openmrs-ocl-client/blob/c3c7d5246006bd0619ee70e947ae2d59a992d392/src/apps/organisations/components/OrgMembers.tsx#L94) where we redner each members user name and, if we are allowed to modify the organisation, the delete button.

So what happens when you click on the delete button? The [`onClick`](https://github.com/openmrs/openmrs-ocl-client/blob/c3c7d5246006bd0619ee70e947ae2d59a992d392/src/apps/organisations/components/OrgMembers.tsx#L100) handler calls `displayDeleteMemberDialog()`. `displayDeleteMemberDialog()` is actually an alias to trigger the [`showDeleteMemberDialogAction`](https://github.com/openmrs/openmrs-ocl-client/blob/c3c7d5246006bd0619ee70e947ae2d59a992d392/src/apps/organisations/components/OrgMembers.tsx#L154). Triggering that action sets the [`showDeleteMemberDialog`](https://github.com/openmrs/openmrs-ocl-client/blob/c3c7d5246006bd0619ee70e947ae2d59a992d392/src/apps/organisations/components/OrgMembers.tsx#L80) property to `true` which shows the `DeleteMemberDialog` (since `showDeleteMemberDialog` is [bound to the `open` property](https://github.com/openmrs/openmrs-ocl-client/blob/c3c7d5246006bd0619ee70e947ae2d59a992d392/src/apps/organisations/components/OrgMembers.tsx#L103)).

Simple enough and it should work. Here's the issue: there isn't just one `DeleteMemberDialog` on the page. There is one _for each member of the organisation_. So which `DeleteMemberDialog` gets show? Apparently the first one. So, no matter which delete button you click on, you are always presented with the `DeleteMemberDialog` for the first member.

Obviously, this needs to be fixed.

The obvious thing to do would be to change `showDeleteMemberDialog` into a variable that can hold a user name so that instead of having a `DeleteMemberDialog` component per member, we have just one.

E.g. something like this:

```ts
<DeleteMemberDialog
  open={showDeleteMemberDialog !== undefined}
  // NB hideDeleteMemberDialog() should now just set showDeleteMember to undefined
  handleClose={hideDeleteMemberDialog}
  handleSubmit={() => deleteMember(orgUrl, showDeleteMember)}
  user={showDeleteMember}
  orgName={orgName}
  error={deleteError}
/>
<List>
  {members?.length ? (
    members.map(m => (
      <ListItem className={classes.root} key={m.username}>
        {m.username || m.name}
        {!canModifyMembers ? null : (
          <>
            <IconButton onClick={() => displayDeleteMemberDialog(m.username)}>
              <DeleteIcon />
            </IconButton>
          </>
        )}
      </ListItem>
    ))
  ) : (
    <li>No members found!</li>
  )}
</List>
```

Reply:
Thank you so much Ian for the detail explanation and finding the bug in product. But I am little confused in the solution above so how should I define showDeleteMember as used in handelSubmit and user?

Reply:
@suruchee Oh... I meant to rename "showDeleteMemberDialog" to "showDeleteMember" (so it sounds less like a boolean value... "showDeleteMember" should be the same value as "showDeleteMemberDialog"

Reply:
I have followed the conversation and I think we need to create a ticket to fix the modal issue, maybe we pass username instead of just a boolean, this might unblock the issue @suruchee is having. Thanks @ibacher for catching that.